src/forms/renderers/DataPlaneSelector/AutoComplete.tsx

```diff
@@ -107,25 +110,29 @@ export const DataPlaneAutoComplete = ({
             onInputChange={(_event, newInputValue) => {
                 setInputValue(newInputValue);
             }}
-            renderGroup={({ group, children }) => (
-                <li key={group}>
-                    <Typography
-                        color="primary"
-                        sx={{
-                            backgroundColor: (theme) =>
-                                theme.palette.background.paper,
-                            fontWeight: 500,
-                            pl: 1,
-                            py: 1,
-                            textTransform: 'capitalize',
-                        }}
-                    >
-                        {intl.formatMessage({ id: `common.${group}` })}
-                    </Typography>
-
-                    <MenuList style={{ padding: 0 }}>{children}</MenuList>
-                </li>
-            )}
+            renderGroup={({ group, children }) =>
+                children === null ||
+                (isArray(children) &&
+                    children.every((node) => node === null)) ? null : (
```

Comment:
This was done to hide group headers when no child nodes can be found. Since the rendered options are filtered, `children` can be null or a null array.

Reply:
This seems backwards though... cause when it isn't true it is going to be checking the entire array every time. I think we should just check if `some` are `not null` instead